java/src/com/mongodb/csfle/CSFLEHelpers.java

```diff
@@ -0,0 +1,219 @@
+                .kmsProviders(kmsProviders)
+                .build();
+
+        return ClientEncryptions.create(clientEncryptionSettings);
+    }
+
+    // You may need to update the following variable to point to your mongocryptd binary
+    private static String mongocryptdPath = "/usr/local/bin/mongocryptd";
+
+    // Creates Encrypted Client which performs automatic encryption and decryption of fields
+    public static MongoClient createEncryptedClient(String connectionString, String kmsProvider, byte[] masterKey, String keyVaultCollection, Document schema, String dataDb, String dataColl) {
+        String recordsNamespace = dataDb + "." + dataColl;
+
+        Map<String, BsonDocument> schemaMap = new HashMap<String, BsonDocument>();
+        schemaMap.put(recordsNamespace, BsonDocument.parse(schema.toJson()));
+
+        Map<String, Object> keyMap = new HashMap<String, Object>();
+        keyMap.put("key", masterKey);
+
+        Map<String, Map<String, Object>> kmsProviders = new HashMap<String, Map<String, Object>>();
+        kmsProviders.put(kmsProvider, keyMap);
+
+        Map<String, Object> extraOpts = new HashMap<String, Object>();
+        extraOpts.put("mongocryptdSpawnPath", mongocryptdPath);
+        // uncomment the following line if you are running mongocryptd manually
+//      extraOpts.put("mongocryptdBypassSpawn", true);
+
+        AutoEncryptionSettings autoEncryptionSettings = AutoEncryptionSettings.builder()
+                .keyVaultNamespace(keyVaultCollection)
+                .kmsProviders(kmsProviders)
+                .extraOptions(extraOpts)
+                .schemaMap(schemaMap)
+                .build();
+
+        MongoClientSettings clientSettings = MongoClientSettings.builder()
+                .applyConnectionString(new ConnectionString(connectionString))
+                .autoEncryptionSettings(autoEncryptionSettings)
+                .build();
+
+        return MongoClients.create(clientSettings);
+    }
+
+    // Returns existing data encryption key
+    public static String findDataEncryptionKey(String connectionString, String keyAltName, String keyDb, String keyColl) {
+        MongoClient mongoClient = createMongoClient(connectionString);
+
+        Document query = new Document("keyAltNames", keyAltName);
+        MongoCollection<Document> collection = mongoClient.getDatabase(keyDb).getCollection(keyColl);
+        BsonDocument doc = collection
+                .withDocumentClass(BsonDocument.class)
+                .find(query)
+                .first();
+
+        if (doc != null) {
+            return Base64.getEncoder().encodeToString(doc.getBinary("_id").getData());
+        }
+        return null;
+    }
+
+    // Creates index for keyAltNames in the specified key collection
+    public static void createKeyVaultIndex(String connectionString, String keyDb, String keyColl) {
+        MongoClient mongoClient = createMongoClient(connectionString);
```

Comment:
If you're creating a MonogClient, you should close it.  You can do it with try-with-resources.

Reply:
Oops, I had read some bad info that `MongoClient` didn't need explicit `close()`. Found the correct information in the latest docs. Will fix.